vm/src/bytesinner.rs

```diff
@@ -1235,3 +1237,95 @@ pub fn bytes_decode(
             ))
         })
 }
+
+fn hex_no_sep(bytes: &[u8]) -> String {
+    let mut buf: Vec<u8> = vec![0; bytes.len() * 2];
+    hex::encode_to_slice(bytes, buf.as_mut_slice()).unwrap();
+    unsafe { String::from_utf8_unchecked(buf) }
+}
+
+pub fn bytes_to_hex(
+    bytes: &[u8],
+    sep: OptionalArg<Either<PyStrRef, PyBytesRef>>,
+    bytes_per_sep: OptionalArg<isize>,
+    vm: &VirtualMachine,
+) -> PyResult<String> {
+    if bytes.is_empty() {
+        return Ok("".to_owned());
+    }
+
```

Comment:
I think seperating arguments processing and main code block would allow to dedent the main code and give visual seperation of the purpose for readers later.

```rust
    let s_guard;
    let b_guard;
    let (sep, bytes_per_sep) = match sep {
        OptionalArg::Present(sep) => {
            let bytes_per_sep = bytes_per_sep.unwrap_or(1);
            if bytes_per_sep == 0 {
                return Ok(hex_no_sep(bytes))
            } 
            let sep = match &sep {
                Either::A(s) => {
                    s_guard = s.borrow_value();
                    s_guard.as_bytes()
                }
                Either::B(bytes) => {
                    b_guard = bytes.borrow_value();
                    b_guard
                }
            };
            (sep, bytes_per_sep)
        }
        OptionalArg::Missing => return Ok(hex_no_sep(bytes)),
    };

    // now main code can be placed here without indent
```